ios/Classes/TiNfcVTagTechnology.m

```diff
@@ -9,4 +9,701 @@
+                                                          blockNumber:[blockNumber integerValue]
+                                                    completionHandler:^(NSError *_Nullable error) {
+                                                      if (![self _hasListeners:@"didExtendedReadMultipleBlocksWithRequestFlags"]) {
+                                                        return;
+                                                      }
+                                                      [self fireEvent:@"didExtendedReadMultipleBlocksWithRequestFlags"
+                                                           withObject:@{
+                                                             @"errorCode" : error != nil ? NUMINTEGER([error code]) : [NSNull null],
+                                                             @"errorDescription" : error != nil ? [error localizedDescription] : [NSNull null],
+                                                             @"errorDomain" : error != nil ? [error domain] : [NSNull null]
+                                                           }];
+                                                    }];
+}
+
+- (void)lockDSFIDWithRequestFlag:(id)args
+{
+  NSNumber *requestFlagsValue = [[args firstObject] valueForKey:@"requestFlags"];
+  uint8_t requestFlags = [requestFlagsValue unsignedCharValue];
+
+  [[self.tagProxy asNFCISO15693Tag] lockDSFIDWithRequestFlag:requestFlags
+                                           completionHandler:^(NSError *_Nullable error) {
+                                             if (![self _hasListeners:@"didLockDSFIDWithRequestFlag"]) {
+                                               return;
+                                             }
+                                             [self fireEvent:@"didLockDSFIDWithRequestFlag"
+                                                  withObject:@{
+                                                    @"errorCode" : error != nil ? NUMINTEGER([error code]) : [NSNull null],
+                                                    @"errorDescription" : error != nil ? [error localizedDescription] : [NSNull null],
+                                                    @"errorDomain" : error != nil ? [error domain] : [NSNull null]
+                                                  }];
+                                           }];
+}
+
+- (void)authenticateWithRequestFlags:(id)args
+{
+  NSNumber *requestFlagsValue = [[args firstObject] valueForKey:@"requestFlags"];
+  uint8_t requestFlags = [requestFlagsValue unsignedCharValue];
+
+  NSNumber *cryptoSuiteIdentifier = [[args firstObject] valueForKey:@"cryptoSuiteIdentifier"];
+
+  NSArray *message = [[args firstObject] valueForKey:@"message"];
+  unsigned int dataValue[message.count];
+  for (int i = 0; i < message.count; i++) {
+    dataValue[i] = [message[i] unsignedIntValue];
+  }
+  NSData *data = [[NSData alloc] initWithBytes:dataValue length:message.count];
```

Comment:
Same here, since it is repetitive code block, better to create a separate methods with name like **arrayToData**.

Reply:
Done